extensions/azurecore/src/extension.ts

```diff
@@ -84,15 +84,27 @@ export async function activate(context: vscode.ExtensionContext) {
 
 // Create the folder for storing the token caches
 async function findOrMakeStoragePath() {
-	let storagePath = path.join(getDefaultLogLocation(), constants.extensionName);
-	try {
-		await fs.mkdir(storagePath, { recursive: true });
-		console.log('Initialized Azure account extension storage.');
-	}
-	catch (e) {
-		console.error(`Initialization of Azure account extension storage failed: ${e}`);
-		console.error('Azure accounts will not be available');
+	let defaultLogLocation = getDefaultLogLocation();
+	let storagePath = path.join(defaultLogLocation, constants.extensionName);
+
+	const stat = await fs.stat(storagePath);
+	if (!stat || !stat.isDirectory()) {
```

Comment:
In general checks like these aren't great since it's a race condition - something could create the directory between this and the call to mkdir. 

What I'd suggest doing is just check the error in the catches and if its EEXIST then just ignore it and continue on.

Reply:
I like what you're saying more than what I've done here! Updating the code.

Thanks for the feedback!